### PR TITLE
fix: `path.parse` to correctly handle absolute paths

### PR DESCRIPTION
**Pull request by privatenumber**
## Problem

Currently `path.parse()` incorrectly returns an empty `root`:
```diff
+ { dir: '/a/b/c',
+   root: '',
+   base: 'index',
+   name: 'index.html',
+   ext: '.html' }
- { root: '/',
-   dir: '/a/b/c',
-   base: 'index.html',
-   ext: '.html',
-   name: 'index' }
```

This causes the https://github.com/sindresorhus/find-up/blob/12605f97f3be2f4ebc4b0d4a9078a165a8680c95/index.js#L55 package, to have an infinite loop because the FS root isn't identified.

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
great bug report

File: test/js/node/path/path.test.js [hunk 1]
```diff
@@ -428,3 +428,40 @@ it("path.resolve", () => {
...
+  describe("absolute path", () => {
+    it("parses", () => {
+      const parsed = path.parse("/a/b/c/index.html");
+      assert.deepStrictEqual(parsed, {
+        root: "/",
+        dir: "/a/b/c",
+        base: "index.html",
+        ext: ".html",
+        name: "index",
+      });
```
Comment (privatenumber) [inline test/js/node/path/path.test.js:448]: This is expected to fail when CI tests on Windows

File: src/bun.js/node/types.zig [hunk 2]
```diff
@@ -1777,17 +1777,22 @@ pub const Path = struct {
...
+                root = JSC.ZigString.init("/");
+            } else {
+                root = JSC.ZigString.init("");
+            }
         }
 
-        var base = JSC.ZigString.init(path_name.base);
-        var name_ = JSC.ZigString.init(path_name.filename);
+        var base = JSC.ZigString.init(path_name.filename);
+        var name_ = JSC.ZigString.init(path_name.base);
```
Comment (privatenumber) [inline src/bun.js/node/types.zig:1801]: This swap is unintuitive. Perhaps the terminology used between `path` and `fs` are inconsistent?

File: src/bun.js/node/types.zig [hunk 3]
```diff
@@ -1777,17 +1777,22 @@ pub const Path = struct {
...
         var dir = JSC.ZigString.init(path_name.dir);
-        if (is_absolute) {
-            root = JSC.ZigString.Empty;
-            if (path_name.dir.len == 0)
-                dir = JSC.ZigString.init(if (isWindows) std.fs.path.sep_str_windows else std.fs.path.sep_str_posix);
+
+        if (isWindows) {
+            if (isZigStringAbsoluteWindows(root)) {
+                root = JSC.ZigString.init(path_name.dir[0..3]);
+            }
```
Comment (dylan-conway) [inline src/bun.js/node/types.zig:1791]: I think `root` on windows also needs to be an empty string when the path is not absolute

Comment (privatenumber) [reply src/bun.js/node/types.zig:1791]: Thanks! Fixed and added relative path tests

**Pull request closed without merging**